### PR TITLE
THC-1366: adding api documentation for validate and optimize j1ql endpoints

### PR DESCRIPTION
**Pull request by angelaco11**

### CODE REVIEW

File: knowledgeBase/APIs/jupiterone-api.md [hunk 1]
```diff
@@ -588,6 +583,73 @@ Example result
 }
 ```
 
+## Validate and Optimize Queries
+
+You can use the validate and optimize j1ql endpoints to validate and/or optimize your queries.
```
Comment (angelaco11) [inline knowledgeBase/APIs/jupiterone-api.md:588]: Let me know if this description can be better or isn't necessary

File: knowledgeBase/APIs/jupiterone-api.md [hunk 2]
```diff
@@ -588,6 +583,73 @@ Example result
...
+    {
+        "query": "Find Person as p that relates to User",
+        "valid": true
+    }
+]
+```
+
+### Optimize queries
+
+The optimize endpoint optimizes and validates a query. It takes in an array of queries and return an array of objects containing the original query, the optimized query with auto-applied optimizations, the list of optimizations that were auto-applied and suggested optimizations that the user can apply, and a valid boolean flag indicating whether the query is valid or not.
```
Comment (benrj) [inline knowledgeBase/APIs/jupiterone-api.md:619]:
```suggestion
The optimize endpoint validates and optimizes a query. It takes in an array of queries and return an array of objects containing the original query, the optimized query, and a list of optimizations that are suggested, and a valid boolean flag indicating whether the query is valid or not.  
```

I don't think the caller needs to worry about the distinction between auto-applied and suggested for this endpoint.  It'll be their job to decide what to do with the optimized query.

Comment (angelaco11) [reply knowledgeBase/APIs/jupiterone-api.md]: Ah yep that makes sense to me. committed your suggestion, thank you!

**Review by benrj — DISMISSED** [review]

**Comment by angelaco11** [comment]
Closing this PR in favor of adding docs to the new docs v2 repo. PR for that here: https://github.com/JupiterOne/docs-v2/pull/112

**Pull request closed without merging**